### ISSUE TITLE
Uses bit shifting even for full-range depth conversions

### ISSUE THREAD

**Issue by Z4ST1N** [EleonoreMizo/fmtconv#10]
for a function call like this:

```
clip = core.std.BlankClip(format=vs.GRAY8, color=255)
clip.fmtc.bitdepth(bits=16, fulls=True, fulld=True).set_output()
```

the clip should have a color of 65535 (255 * 257) but instead is 65280 (255 * 256)

**Comment by mrhso** [comment]
Even RGB.

**Comment by keytouch** [comment]
Vapoursynth resize filter acts in the same way. But theoratically the factor should be 257 to do evenly matched mapping.

**Comment by keytouch** [comment]
In full range it should be 257, when limited it's 256 checked out in vapoursynth.

**Comment by EleonoreMizo** [comment]
Right. Thank you for the report. Fixed in 1695707dec303da59e8eb7d0133c70d804c62190, will be available in upcoming r22 release.
